File: returned/python/merge_k_sorted_lists_test.py
from merge_k_sorted_lists import Solution, to_array, to_nodes, to_node_list

def test1():
    nodes = to_nodes([1,2,3,4,5])
    assert to_array(nodes) == [1,2,3,4,5]

# def test2():
#     lists = to_node_list([[1,4,5],[1,3,4],[2,6]])
#     result = Solution.mergeKLists(None, lists)
#     assert to_array(result) == [1,1,2,3,4,4,5,6]
    
# def test3():
#     lists = to_node_list([])
#     result = Solution.mergeKLists(None, lists)
#     assert to_array(result) == []

# def test4():
#     lists = to_node_list([[]])
#     result = Solution.mergeKLists(None, lists)
#     assert to_array(result) == []

File: csharp/LeetTests/Today4/MergeKSortedListsTest.cs
using Today4.MergeKSortedLists;
namespace Today4;

internal class MergeKSortedListsTest
{
    [Test]
    public void Test1()
    {
        var solution = new Solution();
        var lists = ToLists([[1, 4, 5], [1, 3, 4], [2, 6]]);
        var result = solution.MergeKLists(lists);
        int[] answer = [1, 1, 2, 3, 4, 4, 5, 6];
        Assert.That(LeetList.ToArray(result), Is.EqualTo(answer));
    }

    [Test]
    public void Test2()
    {
        var solution = new Solution();
        var lists = ToLists([]);
        var result = solution.MergeKLists(lists);
        int[] answer = [];
        Assert.That(LeetList.ToArray(result), Is.EqualTo(answer));
    }

    [Test]
    public void Test3()
    {
        var solution = new Solution();
        var lists = ToLists([[]]);
        var result = solution.MergeKLists(lists);
        int[] answer = [];
        Assert.That(LeetList.ToArray(result), Is.EqualTo(answer));
    }

    private ListNode[] ToLists(int[][] lists)
    {
        var result = new List<ListNode>();
        foreach (var list in lists)
        {
            result.Add(LeetList.ToList(list));
        }
        return result.ToArray();
    }
}